Test/Integration/ConfigTest.php

```diff
@@ -236,12 +230,9 @@ public function testExtraSettings()
 
         foreach ($sections as $section) {
             $this->setConfig('algoliasearch_extra_settings/extra_settings/' . $section . '_extra_settings', '{"exactOnSingleWordQuery":"word"}');
-        }
-
-        $indicesConfigurator->saveConfigurationToAlgolia(1);
-        $this->algoliaHelper->waitLastTask();
+            $indicesConfigurator->saveConfigurationToAlgolia(1);
```

Comment:
This seems like a heavy operation to call within the section loop?

Reply:
I understand this makes you react a little bit.
The way it was implemented before doesn't work consistently, I think this is because how the `waitLastTask()` works (by default the waiting is based on the last indexname and the last task id). 
The fact is, the `saveConfigurationToAlgolia()` method updates all the indices (products, categories, pages, suggestions and additional sections) and as we are looping just after, we are not sure that the first checked section index has been successfully updated since the waitLastTask() was waiting for another index (not sure if I'm clear).

BUT, it seems that we have the same problem with the way I implemented it... I think it just works because we let more time to all indices to be updated because it adds more requests ...

Reply:
I decided to switch back to the old implem and add waiting for all indices in the saveConfigurationToAlgolia method, and it seems to work now.

This shouldn't change anything in the method except for that, WDYT ?

Reply:
Yup! That change makes sense. Thanks for explaining the issue!